apm-agent-plugins/apm-spring-webmvc-plugin/src/main/java/co/elastic/apm/agent/spring/webmvc/ViewRenderInstrumentation.java

```diff
@@ -0,0 +1,123 @@
+import co.elastic.apm.agent.impl.transaction.Span;
+import co.elastic.apm.agent.impl.transaction.TraceContextHolder;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.springframework.web.servlet.view.AbstractView;
+import org.thymeleaf.spring4.view.AbstractThymeleafView;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Collection;
+
+import static net.bytebuddy.matcher.ElementMatchers.hasSuperType;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+public class ViewRenderInstrumentation extends ElasticApmInstrumentation {
+
+    private static final String SPAN_TYPE = "template";
+    private static final String SPAN_ACTION = "render";
+    private static final String DISPATCHER_SERVLET_RENDER_METHOD = "View#render";
+
+    @Override
+    public Class<?> getAdviceClass() {
+        return ViewRenderAdviceService.class;
+    }
+
+    public static class ViewRenderAdviceService {
+
+        @Advice.OnMethodEnter(suppress = Throwable.class)
+        public static void beforeExecute(@Advice.Local("span") @Nullable Span span,
+                                         @Advice.This @Nullable Object thiz) {
+            if (tracer == null || tracer.getActive() == null) {
+                return;
+            }
+            final TraceContextHolder<?> parent = tracer.getActive();
+
+            String viewClassName = thiz.getClass().getSimpleName().replace("View", "");
+
+            span = parent.createSpan()
+                .withType(SPAN_TYPE)
+                .withSubtype(viewClassName)
+                .withAction(SPAN_ACTION)
+                .withName(DISPATCHER_SERVLET_RENDER_METHOD);
+
+            String viewName = null;
+            if (thiz instanceof AbstractView) {
+                AbstractView view = (AbstractView) thiz;
+                viewName = view.getBeanName();
+            }
+            switch (viewClassName) {
+                case "Thymeleaf":
+                    AbstractThymeleafView view = (AbstractThymeleafView) thiz;
+                    viewName = view.getBeanName();
+                    break;
+                default:
+                    break;
+            }
+
+            if (viewName != null) {
+                span.appendToName(" ").appendToName(viewName);
+            }
+            span.activate();
+        }
+
+        @Advice.OnMethodExit(suppress = Throwable.class, onThrowable = Throwable.class)
+        public static void afterExecute(@Advice.Local("span") @Nullable Span span,
+                                        @Advice.Thrown @Nullable Throwable t) {
+            if (span != null) {
+                span.captureException(t)
+                    .deactivate()
+                    .end();
+            }
+        }
+    }
+
```

Comment:
add a `getTypeMatcherPreFilter` with `nameContains("View")` to speed up the matching 🙂

Reply:
added